### ISSUE TITLE
PDF's 45º lines dissapearing in png conversion

### ISSUE THREAD

**Issue by gabrielbeneli-missler** [pymupdf/PyMuPDF#3172]
### Description of the bug

Hello,

I'm using the following function to convert PDF files into PNG:

```
        pix = page.get_pixmap(matrix=fitz.Matrix(40, 40),alpha=False,colorspace=fitz.csGRAY)
        pix.save(image_path, "png")
```

 The conversion occurs as expected but every 45º line dissapears:
 
 
![image](https://github.com/pymupdf/PyMuPDF/assets/137560848/7ae0088d-952d-4096-aa95-c8798bf20e92)

 
 
    
    

### How to reproduce the bug

Using this function to convert the attached PDF.

```
def convert_pdf_to_images(pdf_path, output_folder):
    """
    Convert a multiple page PDF into single PNG files

    - pdf_path (str): A string containning the path for the PDF to be converted.
    - output_folder (str): A string containning the path for the folder to dump the images.

    """
    doc = fitz.open(pdf_path)

    for page_num in range(len(doc)):
        page = doc[page_num]
        pagenumiter = ('00000' + str(page_num + 1))[-5:]
        image_name = f"page_{pagenumiter}.png"
        image_path = f"{output_folder}/{image_name}"

        # Convert the PDF page to an image
        pix = page.get_pixmap(matrix=fitz.Matrix(40, 40),alpha=False,colorspace=fitz.csGRAY)
        pix.save(image_path, "png")

        print(f"Converted page {page_num + 1} to {image_path}")

    doc.close()
```
![image](https://github.com/pymupdf/PyMuPDF/assets/137560848/7ae0088d-952d-4096-aa95-c8798bf20e92)

[test.pdf](https://github.com/pymupdf/PyMuPDF/files/14311997/test.pdf)

### PyMuPDF version

1.23.22

### Operating system

Windows

### Python version

3.11

**Comment by JorjMcKie** [comment]
This is an issue in the base library. I will submit a bug report in their system.

**Comment by gabrielbeneli-missler** [comment]
@JorjMcKie thank you for your help!

**Comment by JorjMcKie** [comment]
The MuPDF team has already developed a fix, which will be integrated in one of our next PyMuPDF releases.

**Comment by julian-smith-artifex-com** [comment]
Fixed in 1.24.0.
